framework/aws-kms/aws-kms-hierarchical-keyring.md

```diff
@@ -47,18 +53,40 @@ On initialization, the caller:
 - MUST provide a [Keystore](../branch-key-store.md)
 - MUST provide a [cache limit TTL](#cache-limit-ttl)
 - MUST provide either a Branch Key Identifier or a [Branch Key Supplier](#branch-key-supplier)
-- MAY provide a max cache size
+- MAY provide a [Cache Type](#cache-type)
+- MAY provide a [Partition ID](#partition-id)
 
-On initialization the Hierarchical Keyring MUST initialize a [cryptographic-materials-cache](../local-cryptographic-materials-cache.md) with the configured cache limit TTL and the max cache size.
-If no max cache size is provided, the crypotgraphic materials cache MUST be configured to a
-max cache size of 1000.
+If the Hierarchical Keyring does NOT get a `Shared` cache on initialization,
+it MUST initialize a [cryptographic-materials-cache](../local-cryptographic-materials-cache.md)
+with the user provided cache limit TTL and the entry capacity.
+If no `cache` is provided, a `DefaultCache` MUST be configured with entry capacity of 1000.
 
 ### Cache Limit TTL
 
 The maximum amount of time in seconds that an entry within the cache may be used before it MUST be evicted.
 The client MUST set a time-to-live (TTL) for [branch key materials](../structures.md#branch-key-materials) in the underlying cache.
 This value MUST be greater than zero.
 
+### Cache Type
+
+Sets the cache bound OR the type of cache scoped (created & bound) to the Hierarchical Keyring.
```

Comment:
this sentence is confusing

Reply:
Good point. It sounded different when this was first written. Thanks for bringing this up. This same documentation is in the smithy model for keyrings where the cache input is defined for the Hierarchical Keyring. I updated that and the spec here. Let me know if this updated explanation makes more sense.